pkg/grpc/api.go

```diff
@@ -691,6 +702,105 @@ func (a *api) getStateStore(name string) (state.Store, error) {
 	return a.stateStores[name], nil
 }
 
+func (a *api) getComponent(componentKind string, componentName string) (component interface{}) {
+	switch componentKind {
+	case "state":
+		if statestore := a.stateStores[componentName]; statestore != nil {
+			component = statestore
+		}
+	case "pubsub":
+		if pubsub := a.pubsubAdapter.GetPubSub(componentName); pubsub != nil {
+			component = pubsub
+		}
+	case "secretstores":
+		if secretstore := a.secretStores[componentName]; secretstore != nil {
+			component = secretstore
+		}
+	case "bindings":
+		if inputbinding := a.inputBindings[componentName]; inputbinding != nil {
+			component = inputbinding
+		} else if outputbinding := a.outputBindings[componentName]; outputbinding != nil {
+			component = outputbinding
+		}
+	}
+	return
+}
+
+func (a *api) GetAllComponentsHealthAlpha1(ctx context.Context, in *runtimev1pb.AllComponentsHealthRequest) (*runtimev1pb.AllComponentsHealthResponse, error) {
+	i := 0
```

Comment:
Can `i` be the index in the for loop in line 735?

Reply:
Done